Content.Client/Disposal/Visualizers/DisposalUnitVisualizer.cs

```diff
@@ -5,165 +5,160 @@
-        private string? _stateFlush;
+    [DataField("state_flush", required: true)]
+    [Access(typeof(DisposalUnitVisualizerSystem))]
+    public string? StateFlush;
 
-        [DataField("flush_sound", required: true)]
-        private SoundSpecifier _flushSound = default!;
+    [DataField("flush_sound", required: true)]
+    [Access(typeof(DisposalUnitVisualizerSystem))]
+    public SoundSpecifier FlushSound = default!;
 
-        [DataField("flush_time", required: true)]
-        private float _flushTime;
+    [DataField("flush_time", required: true)]
+    [Access(typeof(DisposalUnitVisualizerSystem))]
+    public float FlushTime;
 
-        private Animation _flushAnimation = default!;
+    [ViewVariables(VVAccess.ReadOnly)]
+    [Access(typeof(DisposalUnitVisualizerSystem))]
+    public Animation FlushAnimation = default!;
+}
 
-        void ISerializationHooks.AfterDeserialization()
-        {
-            _flushAnimation = new Animation {Length = TimeSpan.FromSeconds(_flushTime)};
+public sealed class DisposalUnitVisualizerSystem : VisualizerSystem<DisposalUnitVisualizerComponent>
+{
+    #region Dependencies
+    [Dependency] private readonly SpriteSystem _spriteSystem = default!;
+    #endregion Dependencies
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<DisposalUnitVisualizerComponent, ComponentInit>(OnInit);
+    }
 
-            var flick = new AnimationTrackSpriteFlick();
-            _flushAnimation.AnimationTracks.Add(flick);
-            flick.LayerKey = DisposalUnitVisualLayers.Base;
-            flick.KeyFrames.Add(new AnimationTrackSpriteFlick.KeyFrame(_stateFlush, 0));
+    private void OnInit(EntityUid uid, DisposalUnitVisualizerComponent comp, ComponentInit args)
+    {
+        comp.FlushAnimation = new Animation {Length = TimeSpan.FromSeconds(comp.FlushTime)};
 
-            var sound = new AnimationTrackPlaySound();
-            _flushAnimation.AnimationTracks.Add(sound);
+        var flick = new AnimationTrackSpriteFlick();
+        comp.FlushAnimation.AnimationTracks.Add(flick);
+        flick.LayerKey = DisposalUnitVisualLayers.Base;
+        flick.KeyFrames.Add(new AnimationTrackSpriteFlick.KeyFrame(comp.StateFlush, 0));
 
-            sound.KeyFrames.Add(new AnimationTrackPlaySound.KeyFrame(_flushSound.GetSound(), 0));
-        }
+        var sound = new AnimationTrackPlaySound();
+        comp.FlushAnimation.AnimationTracks.Add(sound);
+
+        sound.KeyFrames.Add(new AnimationTrackPlaySound.KeyFrame(comp.FlushSound.GetSound(), 0));
+    }
 
-        private void ChangeState(AppearanceComponent appearance)
+    protected override void OnAppearanceChange(EntityUid uid, DisposalUnitVisualizerComponent comp, ref AppearanceChangeEvent args)
+    {
+        var sprite = args.Sprite;
+        if (sprite == null)
+            return;
+        if (!AppearanceSystem.TryGetData(uid, Visuals.VisualState, out VisualState state, args.Component))
```

Comment:
make sure to always use the generic version `TryGetData<>()` so the cast is done in the method